src/aliceVision/sfmData/colorize.cpp

```diff
@@ -31,97 +62,52 @@ void colorizeTracks(SfMData& sfmData)
   for(auto& landmarkPair : sfmData.getLandmarks())
     remainingLandmarksToColor.push_back(landmarkPair.second);
 
-  struct ViewInfo
-  {
-    ViewInfo(IndexT viewId, std::size_t cardinal)
-      : viewId(viewId)
-      , cardinal(cardinal)
-    {}
-
-    IndexT viewId;
-    std::size_t cardinal;
-    std::vector<std::reference_wrapper<Landmark>> landmarks;
-  };
-
-  std::vector<ViewInfo> sortedViewsCardinal;
-  sortedViewsCardinal.reserve(sfmData.getViews().size());
-  {
-    // create cardinal per viewId map
-    std::map<IndexT, std::size_t> viewsCardinalMap; // <ViewId, Cardinal>
-    for(const auto& landmarkPair : sfmData.getLandmarks())
-    {
-      const Observations& observations = landmarkPair.second.observations;
-      for(const auto& observationPair : observations)
-        ++viewsCardinalMap[observationPair.first]; // TODO: 0
-    }
-
-    // copy key-value pairs from the map to the vector
-    for(const auto& cardinalPair : viewsCardinalMap)
-      sortedViewsCardinal.push_back(ViewInfo(cardinalPair.first, cardinalPair.second));
-
-    // sort the vector, biggest cardinality first
-    std::sort(sortedViewsCardinal.begin(),
-              sortedViewsCardinal.end(),
-              [] (const ViewInfo& l, const ViewInfo& r) { return l.cardinal > r.cardinal; });
-  }
+  std::vector<LMColorAccumulator> landmarkInfo(sfmData.getLandmarks().size());
+  const Views& views = sfmData.getViews();
 
-  // assign each landmark to a view
-  for(ViewInfo& viewCardinal : sortedViewsCardinal)
+  #pragma omp parallel for
+  for( int viewIndex = 0; viewIndex < views.size(); ++viewIndex)
   {
-    std::vector<std::reference_wrapper<Landmark>> toKeep;
-    const IndexT viewId = viewCardinal.viewId;
+      image::Image<image::RGBColor> image;
 
-    for(int i = 0; i < remainingLandmarksToColor.size(); ++i)
-    {
-      Landmark& landmark = remainingLandmarksToColor.at(i);
-      auto it = landmark.observations.find(viewId);
-      if(it != landmark.observations.end())
-      {
-        viewCardinal.landmarks.push_back(landmark);
-      }
-      else
+      auto itView = views.begin();
+      std::advance(itView, viewIndex);
+      const IndexT viewId = itView->first;
+      image::readImage(itView->second->getImagePath(), image, image::EImageColorSpace::SRGB);
+
+      #pragma omp parallel for
+      for(int i = 0; i < remainingLandmarksToColor.size(); ++i)
       {
-        toKeep.push_back(landmark);
+            const Landmark& landmark = remainingLandmarksToColor.at(i);
+            const auto it = landmark.observations.find(viewId);
+
+            if(it != landmark.observations.end())
+            {
+                const Vec3& Tcenter = sfmData.getAbsolutePose(viewId).getTransform().center();
+                const Vec3& pt = landmark.X;
+                const double eucd = 1.0 / (Tcenter - pt).norm();
```

Comment:
This might throw a division by zero error, you should either add a small constant or use any other formula that ensures this division is safe, for example `const double eucd = 1.0 / std::max((Tcenter - pt).norm(), 1e-5);`

Reply:
Done